app/models/document/paginated_history.rb

```diff
@@ -0,0 +1,65 @@
+class Document::PaginatedHistory
+  PER_PAGE = 30
+
+  attr_reader :document, :query
+
+  delegate :total_count, to: :query
+
+  def initialize(document, page)
+    @document = document
+    @query = document.edition_versions
+                     .includes(:user)
+                     .where.not(state: :superseded)
+                     .reorder(created_at: :desc, id: :desc)
+                     .page(page)
+                     .per(PER_PAGE)
+  end
+
+  def audit_trail
+    first_edition_id = document.first_edition&.id
+    query.map.with_index do |version, index|
+      # [index - 1] returns next version, as array has newest first
```

Comment:
I don't think this comment is helpful any more. I think we should probably just remove it to avoid future confusion. _(e.g. "Why does the comment say `index - 1` when the code does `index + 1`? 🤔")_

```suggestion
```